src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 10;
+    private int size;
+
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[])(new Object[DEFAULT_ARRAYLIST_SIZE]);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 > array.length) {
+            growArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        outOfRangeCheck((index == size && size != 0) ? (index - 1) : index);
+        if (size + 1 > array.length) {
+            growArray();
+        }
+        System.arraycopy(array, index, array, index + 1,array.length - index - 1);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > array.length) {
+            growArray();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        outOfRangeCheck(index == size ? index + 1 : index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        outOfRangeCheck(index == size ? index + 1 : index);
+        array[index] = value;
+        if (index > size) {
+            size = index;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        outOfRangeCheck(index == size ? index + 1 : index);
```

Comment:
```suggestion
        checkIndex(index);
```
this ternary operator is not easy to understand here, and not needed